src/core/layertree/qgslayertree.cpp

```diff
@@ -54,31 +55,49 @@ void QgsLayerTree::setCustomLayerOrder( const QStringList &customLayerOrder )
 {
   QList<QgsMapLayer *> layers;
 
-  Q_FOREACH ( const QString &layerId, customLayerOrder )
+  for ( const auto &layerId : customLayerOrder )
   {
     QgsLayerTreeLayer *nodeLayer = findLayer( layerId );
     if ( nodeLayer )
     {
+      // configuration from 2.x projects might have non spatial layers
+      QgsMapLayer *layer = nodeLayer->layer();
+      if ( layer->type() == QgsMapLayer::VectorLayer )
```

Comment:
QgsMapLayer has isSpatial (always true for rasters), so you could avoid all the casting to vectors here.

Reply:
thanks for the reminder!
fixed